src/core/utils/urlutils.js

```diff
@@ -82,3 +82,19 @@ export function equivalentParams (params1, params2) {
   }
   return true;
 }
+
+/**
+ * Removes params that begin with the given prefixes
+ * Note: modifies the params parameter
+ * @param {SearchParams} params The parameters to remove from
+ * @param {string[]} prefixes The prefixes of parameters to remove
+ */
+export function removeParamsWithPrefixes (params, prefixes) {
```

Comment:
Instead of mutating the params object, could we create a copy and update that one? I always prefer to avoid mutation when possible (and when I remember, haha).

Reply:
Done, had it like that before but I was 50/50 on balancing

1. time to create a new SearchParams for every call
2. keeping function pure

Found a way to do it without have to copy over the array before doing any filtering